auth/store.go

```diff
@@ -1090,7 +1090,9 @@ func (as *authStore) WithRoot(ctx context.Context) context.Context {
 		"token": token,
 	}
 	tokenMD := metadata.New(mdMap)
-	return metadata.NewOutgoingContext(ctx, tokenMD)
+
+	// use "mdIncomingKey{}" since it's called from local etcdserver
+	return metadata.NewIncomingContext(ctx, tokenMD)
```

Comment:
Funny, I was just reading about these new grpc metadata context functions for the first time today.

Reply:
Yeah, it was from gRPC v1.3.0 with https://github.com/grpc/grpc-go/pull/1157.